### PR TITLE
Batch db write in order to resolve test slowliness

### PR DESCRIPTION
**Pull request by shayzluf**
[Resolves] #1714 

---

# Testing validators was taking ~400s batching help to reduce it to ~1s

According to the issue the testing time was an issue that could be resolved by batching. So here I try to resolve it

Added batch function to db.go
added SaveValidatorIndexBatch function to validator.go

### CODE REVIEW

File: beacon-chain/db/validator.go [hunk 1]
```diff
@@ -23,6 +23,20 @@ func (db *BeaconDB) SaveValidatorIndex(pubKey []byte, index int) error {
...
 
+// SaveValidatorIndexBatch accepts a public key and validator index and writes them to disk.
+func (db *BeaconDB) SaveValidatorIndexBatch(pubKey []byte, index int) error {
+	h := hashutil.Hash(pubKey)
+
+	return db.batch(func(tx *bolt.Tx) error {
+		bucket := tx.Bucket(validatorBucket)
+		buf := make([]byte, binary.MaxVarintLen64)
+		n := binary.PutUvarint(buf, uint64(index))
+		err := bucket.Put(h[:], buf[:n])
```
Comment (nisdas) [inline beacon-chain/db/validator.go]: You can just do `return bucket.Put(h[:], buf[:n])`

File: beacon-chain/rpc/validator_server_test.go [hunk 2]
```diff
@@ -182,11 +183,22 @@ func TestNextEpochCommitteeAssignment_OK(t *testing.T) {
...
-	for i := 0; i < int(params.BeaconConfig().DepositsForChainStart); i++ {
+	var wg sync.WaitGroup
+	numOfValidators := int(params.BeaconConfig().DepositsForChainStart)
+	errs := make(chan error, numOfValidators)
+	for i := 0; i < numOfValidators; i++ {
 		pubKeyBuf := make([]byte, binary.MaxVarintLen64)
 		n := binary.PutUvarint(pubKeyBuf, uint64(i))
-		if err := db.SaveValidatorIndex(pubKeyBuf[:n], i); err != nil {
+		wg.Add(1)
+		go func(itemID string) {
```
Comment (nisdas) [inline beacon-chain/rpc/validator_server_test.go]: so you dont use `itemID` here, so not sure what is the use case. Why use strings when indexes are `int` ?

Comment (shayzluf) [reply beacon-chain/rpc/validator_server_test.go]: my mistake

Comment (shayzluf) [reply beacon-chain/rpc/validator_server_test.go]: fixed it

**Pull request merged**